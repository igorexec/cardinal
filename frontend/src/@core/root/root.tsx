import React from 'react';
import {BrowserRouter, Switch, Route, Redirect} from 'react-router-dom';
import {Header} from '@core/header';
import {Footer} from '@core/footer';
import {Dashboard, PageSpeed} from '@pages';

export const Root: React.FC = () => {
  const navItems = [{
    id: '1', link: '/pagespeed', title: 'PageSpeed',
  }, {
    id: '2', link: '/crawl', title: 'Crawl Issues',
  }];

  return (
    <BrowserRouter>
      <Header navItems={navItems} />
      <Switch>
        <Route exact path="/" component={Dashboard} />
        <Route path="/pagespeed" component={PageSpeed} />
        <Redirect to="/" />
      </Switch>
      <Footer />
    </BrowserRouter>
  );
};
