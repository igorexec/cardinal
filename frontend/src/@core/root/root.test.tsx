import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import {Root} from './root';

describe('Root', () => {
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    wrapper = shallow(<Root />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
