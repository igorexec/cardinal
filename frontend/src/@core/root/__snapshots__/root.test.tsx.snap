// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root matches snapshot 1`] = `
<BrowserRouter>
  <Header
    navItems={
      Array [
        Object {
          "id": "1",
          "link": "/pagespeed",
          "title": "PageSpeed",
        },
        Object {
          "id": "2",
          "link": "/crawl",
          "title": "Crawl Issues",
        },
      ]
    }
  />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/pagespeed"
    />
    <Redirect
      to="/"
    />
  </Switch>
  <Footer />
</BrowserRouter>
`;
